shell/components/form/FileSelector.vue

```diff
@@ -63,7 +77,13 @@ export default {
     accept: {
       type:    String,
       default: '*'
-    }
+    },
+
+    fileType: {
```

Comment:
This can be covered using the existing `accept` prop to restrict which file types can be selected for upload:  https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#limiting_accepted_file_types. I would add an accept prop to the `FileImageSelector` component that defaults to allowing all images: 
```
 accept: {
    type:    String,
    default: 'image/*'
  }
```

Then pass through more specific restrictions in the navlinks component: 
```
   <FileImageSelector
     v-model="value.spec.iconSrc"
     :read-as-data-url="true"
     :mode="mode"
     :label="t('navLink.tabs.groupImage.iconSrc.label')"
     accept="image/jpeg,image/png,image/svg+xml"
     @error="setImageError"
     @input="setIcon"
   />